common/components/ui/Modal.jsx

```diff
@@ -0,0 +1,107 @@
+// @flow
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import closeIcon from 'assets/images/icon-x.svg';
+
+import './Modal.scss';
+
+export default class Modal extends Component {
+  static propTypes = {
+    isOpen: PropTypes.bool,
+    size: PropTypes.oneOf(['small', 'medium', 'large', 'xlarge']),
+    title: PropTypes.node.isRequired,
+    children: PropTypes.node,
+    buttons: PropTypes.arrayOf(
+      PropTypes.shape({
+        text: PropTypes.node.isRequired,
+        type: PropTypes.oneOf([
+          'default',
+          'primary',
+          'success',
+          'info',
+          'warning',
+          'danger',
+          'link'
+        ]),
+        onClick: PropTypes.func.isRequired
+      })
+    ),
+    handleClose: PropTypes.func.isRequired
+  };
+
+  componentDidMount() {
+    document.addEventListener('keydown', this._escapeListner);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keydown', this._escapeListner);
+  }
+
+  _escapeListner = ev => {
+    // Don't trigger if they hit escape while on an input
+    if (ev.target) {
+      if (
+        ev.target.tagName === 'INPUT' ||
+        ev.target.tagName === 'SELECT' ||
+        ev.target.tagName === 'TEXTAREA' ||
+        ev.target.isContentEditable
+      ) {
+        return;
+      }
+    }
+
+    if (ev.key === 'Escape' || ev.keyCode === 27) {
+      this.props.handleClose();
+    }
+  };
+
+  _renderButtons() {
+    let idx = 0;
+
+    return this.props.buttons.map(btn => {
```

Comment:
How have I gone on this long without knowing that map had a second argument for index...